packages/itwin/breakdown-trees/src/Views/FunctionalityProviders/SelectRelatedFunctionalityProvider.ts

```diff
@@ -3,21 +3,20 @@
 * See LICENSE.md in the project root for license terms and full copyright notice.
 *--------------------------------------------------------------------------------------------*/
 
-
-import { KeySet, NodeKey } from "@bentley/presentation-common";
-import { IPresentationTreeDataProvider } from "@bentley/presentation-components";
-import { Presentation } from "@bentley/presentation-frontend";
-import { TreeModelNode, TreeNodeItem, DelayLoadedTreeNodeItem } from "@bentley/ui-components";
-import { IModelConnection } from "@bentley/imodeljs-frontend";
+import { KeySet, NodeKey } from "@itwin/presentation-common";
+import type { IPresentationTreeDataProvider } from "@itwin/presentation-components";
+import { Presentation } from "@itwin/presentation-frontend";
+import type { TreeModelNode, TreeNodeItem } from "@itwin/components-react";
+import type { IModelConnection } from "@itwin/core-frontend";
 import { TreeNodeFunctionalityProvider } from "./TreeNodeFunctionalityProvider";
 import { RelatedElementIdsProvider } from "../RelatedElementIdsProvider";
-import { BeEvent, Id64String } from "@bentley/bentleyjs-core";
+import type { BeEvent, Id64String } from "@itwin/core-bentley";
 
 export class SelectRelatedFunctionalityProvider extends TreeNodeFunctionalityProvider {
   private _rulesetId: string;
-  private _onActionPerformedEvent: BeEvent<() => void>;
+  private _onActionPerformedEvent: BeEvent<() => void> | undefined;
```

Comment:
why is this potentially undefined?

Reply:
Yeah, this one can be any additional event handler user wants to execute, like from factorysight we use it for tracking in appInsights.. So it can be undefined if user does not need any additional functionality like this.  Previously when it was a mandate, we were handling the call with !, but with eslint its throwing error @typescript-eslint/no-non-null-asserted-optional-chain. So decided to allow undefined here.